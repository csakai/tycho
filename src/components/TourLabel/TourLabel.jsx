import React from 'react';

export default class TourLabel extends React.Component {
  render() {
    return (
      <div className="tour-label">
        <span className={`
          tour-label__text
          tour-label__text--${this.props.modifier}`
        }>
          {this.props.text}
        </span>
      </div>
    );
  }
}

