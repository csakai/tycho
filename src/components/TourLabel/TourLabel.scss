.tour-label {
  position: absolute;
  top: 0;
  width: 100%;
  text-align: center;
  
  &__text {
    background-color: rgba(0, 0, 0, 0.5);
    color: #fff;
    padding: 1em;
    font-family: Georgia, serif;
    transition: opacity 0.5s;
    opacity: 0;

    &--show {
      opacity: 1;
    }

    &--hide {
      opacity: 0;
    }
  }
}
