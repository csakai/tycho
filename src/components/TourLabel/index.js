import TourLabel from './TourLabel';
import './TourLabel.css';

export default TourLabel;
