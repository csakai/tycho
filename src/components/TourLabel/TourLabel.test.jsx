import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import TourLabel from './TourLabel';

describe('TourLabel Component', () => {
  describe('render()', () => {
    it('should render the app successfully', () => {
      const component = shallow(
        <TourLabel
          modifier="hide"
          text="Hello, world"
        />
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
