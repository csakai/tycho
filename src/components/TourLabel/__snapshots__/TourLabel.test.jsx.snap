// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourLabel Component render() should render the app successfully 1`] = `
<div
  className="tour-label"
>
  <span
    className="
          tour-label__text
          tour-label__text--hide"
  >
    Hello, world
  </span>
</div>
`;
